azimuth/modules/base_classes/caching.py

```diff
@@ -101,6 +107,10 @@ def _store_data_in_cache(self, result: List[ModuleResponse], indices: List[int])
                 ds[()] = arr
             handle.flush()
 
+        # Save all that affects caching so it can be used for identification and debugging.
+        with open(pjoin(self._cache_metadata), "w+") as f:
```

Comment:
Detail, but I always hated that habit of renaming `pjoin`. I prefer
```diff
-from os.path import join as pjoin
+import os
```
```suggestion
        with open(os.path.join(self._cache_metadata), "w+") as f:
```

Reply:
Second comment on that line.
This syntax is a nightmare, but the difference between `w` and `w+` is that `w+` gives the ability to read `f`. This is not necessary here, so we can remove the `w`.
```suggestion
        with open(os.path.join(self._cache_metadata), "w") as f:
```
I found a good diagram explaining the syntax, where `truncate` essentially means "empty the file".
![](https://d20p74l5mne5au.cloudfront.net/ExWNT-white-bg.png)

Reply:
Oh awesome, thank you!! Very helpful!